core/unit_test/TestMDSpanConversion.hpp

```diff
@@ -120,12 +120,15 @@ struct TestViewMDSpanConversion {
     }
     // test conversion operator to different mdspan type
     {
+      using const_acc_type = Kokkos::Impl::SpaceAwareAccessor<
+          typename ViewType::memory_space,
+          Kokkos::default_accessor<
+              const typename natural_mdspan_type::element_type>>;
       using mdspan_type = Kokkos::mdspan<
           const typename natural_mdspan_type::element_type,
           Kokkos::dextents<typename natural_mdspan_type::index_type,
                            natural_mdspan_type::rank()>,
-          typename natural_mdspan_type::layout_type,
-          typename natural_mdspan_type::accessor_type>;
+          typename natural_mdspan_type::layout_type, const_acc_type>;
```

Comment:
Isn't this the same bug fixed by https://github.com/kokkos/kokkos/pull/7070

We may want to have a more general solution if we add other accessor types

Reply:
yeah I rebased now to have this all sorted out